executor-service/src/main/java/executor/service/service/proxy/ProxySourcesClientService.java

```diff
@@ -0,0 +1,59 @@
+package executor.service.service.proxy;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import executor.service.model.ProxyConfigHolder;
+import executor.service.model.ProxyCredentials;
+import executor.service.model.ProxyNetworkConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProxySourcesClientService implements ProxySourcesClient {
+    private static final String PROXY_CREDENTIALS_PATH = "src/json/ProxyCredentials.json";
+    private static final String PROXY_NETWORK_CONFIG_PATH = "src/json/ProxyNetworkConfig.json";
+    private final ObjectMapper objectMapper;
+
+    public ProxySourcesClientService(ObjectMapper objectMapper) throws IOException {
+        this.objectMapper = objectMapper;
+    }
+
+    private List<ProxyCredentials> getProxyCredentialsFromFile() throws IOException {
+        return objectMapper.readValue(
+                new File(PROXY_CREDENTIALS_PATH),
+                objectMapper.getTypeFactory()
+                        .constructCollectionType(List.class, ProxyCredentials.class));
+    }
+
+    private List<ProxyNetworkConfig> getProxyNetworkConfigFromFile() throws IOException {
+        return objectMapper.readValue(
+                new File(PROXY_NETWORK_CONFIG_PATH),
+                objectMapper.getTypeFactory()
+                        .constructCollectionType(List.class, ProxyNetworkConfig.class));
+    }
+
+    @Override
+    public ProxyConfigHolder getProxy() throws IOException {
```

Comment:
Each time getProxy() is called, both files are read.
If it is a proxy, you can divide some cache for ProxyConfigHolder
A proxy template implemented on RealProxySourcesClientService is also a good approach